pkg/bigint/bigint.go

```diff
@@ -0,0 +1,37 @@
+package bigint
+
+import (
+	"encoding/json"
+	"fmt"
+	"math/big"
+)
+
+type BigInt struct {
+	big.Int
+}
+
+func (i BigInt) MarshalJSON() ([]byte, error) {
+	return []byte(fmt.Sprintf(`"%s"`, i.String())), nil
+}
+
+func (i *BigInt) UnmarshalJSON(b []byte) error {
+	var val string
+	err := json.Unmarshal(b, &val)
+	if err != nil {
+		return err
+	}
+
+	i.SetString(val, 10)
+
+	return nil
+}
+
+func NewBigInt(x int64) *BigInt {
```

Comment:
since package name is already `bigint`, no need to repeat the name again.
compare `bigint.New(123)`  with `bigint.NewBigInt(123)`

Reply:
Also, I went through the rest of the CL, and it seems that this method is only used in tests. I would therefore remove it and just initialize a `big.Int` in the tests, and just call `Wrap` in case it is needed